cmd/auth/add.go

```diff
@@ -58,7 +58,7 @@ var addCmd = &cobra.Command{
 		// search for provider with same name
 		providerIndex := -1
 		for i, provider := range configAI.Providers {
-			if backend == provider.Name {
+			if backend == provider.Name || defaultBackend == provider.Name{
```

Comment:
In my opinion, instead of adding another check we should restructure the function to validate the input upfront. So the general flow of the updated function should go like:

```
1. Get ai configuration
2. Validate input values upfront and set default values if the inputs are empty
3. Find existing provider index
4. Instantiate the newProvider object.
5. Add the provider based on the value of providerIndex.
```

Reply:
Yeah, that's also a great approach. If @AlexsJones wants to fix this issue in that way, I'll go with that approach.

Reply:
I agree with @VaibhavMalik4187 lets bias towards readability/maintainability here

Reply:
Okay I will make changes and update this PR.